client/src/store/index.js

```diff
@@ -218,16 +218,18 @@ const store = new Vuex.Store({
       log('Down to %o publications after filters', basePublications.length)
 
       if (search) {
-        const fuse = new Fuse(basePublications, fuseOptions, index)
+        const useIndex = isEmpty(filters) ? index : null
+        const fuse = new Fuse(basePublications, fuseOptions, useIndex)
```

Comment:
what’s the benefit of skipping the index?

Reply:
no benefit, but if the publications were filtered then the index becomes invalid, since it references the publications by position in the array. There are probably better solutions, but for me it doesn't seem like the index brings much performance benefit.

Reply:
```js
const Fuse = require('fuse.js')
l = ["abc1", "bcd1", "cdef", "xzy"]
index = Fuse.createIndex({}, l)
l2 = l.filter(x => !x.includes("1"))

// without index
new Fuse(l, {}).search("def") // [ { item: 'cdef', refIndex: 2 } ]
new Fuse(l2, {}).search("def") // [ { item: 'cdef', refIndex: 0 } ]


// with index
new Fuse(l, {}, index).search("def") // [ { item: 'cdef', refIndex: 2 } ]
new Fuse(l2, {}, index).search("def") // [ { item: undefined, refIndex: 2 } ]
```